plugin-flex-ts-template-v2/src/feature-library/agent-automation/flex-hooks/events/taskReceived.ts

```diff
@@ -0,0 +1,50 @@
+import * as Flex from "@twilio/flex-ui";
+import { ITask } from "@twilio/flex-ui";
+import { isFeatureEnabled, getMatchingTaskConfiguration } from "../.."
+import { TaskQualificationConfig } from "feature-library/agent-automation/types/ServiceConfiguration";
+
+
+function selectAndAcceptTask(task: ITask, taskConfig: TaskQualificationConfig) {
+  const {
+    sid
+  } = task
+
+  // Attempt to accept the task
+  if(taskConfig.auto_accept) Flex.Actions.invokeAction("AcceptTask", { sid });
+
+  // Creating an interval to check the task has been accepted
+  // and retries if it has not.  To avoid a runway process,
+  // applying an attempts counter to ultimately abandon the retry
+  // after ten attempts
+  if(taskConfig.auto_select && taskConfig.auto_accept) {
+    let attempts = 0;
+    const selectTaskTimer = setInterval(() => {
+      const task = Flex.TaskHelper.getTaskByTaskSid(
+        sid
+      );
+      if (!task || !task.status) {
+        clearInterval(selectTaskTimer);
+        attempts = 0;
+      } else if (Flex.TaskHelper.isTaskAccepted(task)) {
+        Flex.Actions.invokeAction("SelectTask", { sid });
+        clearInterval(selectTaskTimer);
+        attempts = 0;
+      } else if (attempts > 10) {
+        clearInterval(selectTaskTimer);
+        attempts = 0;
+      } else {
+        attempts++;
+      }
+    }, 500);
+  } else if (taskConfig.auto_select) {
+    Flex.Actions.invokeAction("SelectTask", { sid });
+  }
```

Comment:
I'd like to learn more about why this is needed, I wasn't able to reproduce any issues by only running SelectTask then AcceptTask, so I must be missing something!

Reply:
looks like using awaits works now 👍